cypress/e2e/pages/visiumTransfer.cy.ts

```diff
@@ -213,11 +213,11 @@ describe('Transfer Page', () => {
         selectOption('bioState', 'Probes');
         cy.findByTestId('copyMode-Many to one').click({ force: true });
         cy.get('#inputLabwares').within(() => {
-          cy.findByText('A2').click({ force: true, timeout: 1000 });
-          cy.findByText('B2').click({ cmdKey: true, force: true, timeout: 1000 });
+          cy.findByText('A2').wait(1000).click({ force: true });
+          cy.findByText('B2').wait(1000).click({ cmdKey: true, force: true });
         });
         cy.get('#outputLabwares').within(() => {
-          cy.findByText('D1').click({ force: true, timeout: 1000 });
+          cy.findByText('D1').wait(1000).click({ force: true });
         });
       });
       it('should display the many to one mode', () => {
```

Comment:
have been dealing with failing Cypress tests for the past few days. The main issues that have been recurring randomly are related to the Visium transfer pages and selecting options from the CustomReactSelect component. This unpredictable behavior seems to be tied to Cypress being unable to complete user events, particularly in pages like Visium transfer, where there is a lot of user interaction, such as selecting slots in both the source and destination.

Upon extensive research, I've found that others are facing similar issues. Therefore, I've added a wait in the Visium transfer page and while selecting options in CustomReactSelect. The dilemma lies between tolerating the failing test behavior in CI, which can be overcome by repeatedly running the failing tests, or introducing a slight delay or wait between actions to ensure these tests pass on the first run. I've chosen the latter option.

Although adding a wait is not ideal, the justification is that it's a better alternative than rerunning the entire Cypress test suite repeatedly to make it pass. I've also attempted to make some performance updates in the SlotMapper component to prevent unnecessary re-rendering, which could be another cause of the failing tests.

I hope this explanation makes sense.

Reply:
Thank you for the explanation. I agree that it's better to add some timeout rather than rerunning the Cypress tests unnecessarily. Probably, when we have time, we should take a closer look at the CustomReactSelect component to see if there are any improvements we can make.

Reply:
Sure 👍